src/Vault.sol

```diff
@@ -546,13 +540,14 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
    */
   function sponsorWithPermit(
     uint256 _assets,
+    address _owner,
     uint256 _deadline,
     uint8 _v,
     bytes32 _r,
     bytes32 _s
   ) external returns (uint256) {
-    _permit(IERC20Permit(asset()), msg.sender, address(this), _assets, _deadline, _v, _r, _s);
-    return _sponsor(_assets);
+    _permit(IERC20Permit(asset()), _owner, address(this), _assets, _deadline, _v, _r, _s);
+    return _sponsor(_assets, _owner);
```

Comment:
This makes sense, because the owner is the one signing off their own sponsorship.